src/org/authInfo.ts

```diff
@@ -419,7 +419,48 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
             logger.debug(`error updating auth file for ${orgAuthInfo.getUsername()}`, error);
           }
         } catch (error) {
-          logger.error(`Error connecting to devhub ${hubAuthInfo.username}`, error);
+          if (error instanceof Error && error.name === 'SingleRecordQuery_NoRecords' && fields.orgId) {
+            // Not a scratch org owned by this hub. Check if it's a sandbox.
+            try {
+              const prodOrg = await Org.create({ aliasOrUsername: hubAuthInfo.username });
+              const sbxProcess = await prodOrg.querySandboxProcessByOrgId(fields.orgId);
+              logger.debug(`${fields.orgId} is a sandbox of ${hubAuthInfo.username}`);
+
+              try {
+                await orgAuthInfo.save({
+                  ...fields,
+                  isScratch: false,
+                  isSandbox: true,
+                });
+              } catch (err) {
+                logger.debug(`error updating auth file for: ${orgAuthInfo.getUsername()}`, err);
+              }
+
+              try {
+                // set the sandbox config value
+                const sfSandbox = {
+                  sandboxUsername: fields.username,
+                  sandboxOrgId: fields.orgId,
+                  prodOrgUsername: hubAuthInfo.username,
+                  sandboxName: sbxProcess.SandboxName,
+                  sandboxProcessId: sbxProcess.Id,
+                  sandboxInfoId: sbxProcess.SandboxInfoId,
+                  timestamp: new Date().toISOString(),
+                } as SandboxFields;
+
+                const stateAggregator = await StateAggregator.getInstance();
+                stateAggregator.sandboxes.set(fields.orgId, sfSandbox);
+                logger.debug(`writing sandbox auth file for: ${orgAuthInfo.getUsername()} with ID: ${fields.orgId}`);
+                await stateAggregator.sandboxes.write(fields.orgId);
```

Comment:
this seems safe from the ConfigFile bugs because it should only ever hit one of these.

Just be sure that the auth command is calling this `identifyPossibleScratchOrgs` NOT in parallel with anything else that might be touching the authFile.

Reply:
I checked all places I could find and nothing was done in parallel so hopefully it's ok.